### ISSUE TITLE
async methods in es6 show parse error instead of suggesting es7 option

### ISSUE THREAD

**Issue by pornel** [eslint/eslint#8201]
**Tell us about your environment**

* **ESLint Version:** v3.17.0
* **Node Version:** v7.7.1
* **npm Version:** 4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
module.exports = {
    "env": {
        "browser": true,
        "commonjs": true,
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "rules": {
        "indent": [
            "error",
            4
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
            "always"
        ]
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
class Foo {
    async bar() {}
}
```

**What did you expect to happen?**
Should parse the file as valid

**What actually happened? Please include the actual, raw output from ESLint.**
> 2:11  error  Parsing error: Unexpected token bar
> ✖ 1 problem (1 error, 0 warnings)

**Comment by mysticatea** [comment]
Thank you for this issue.

However, async functions are not a part of ES6 (ES2015).
Use `parserOptions` with `ecmaVersion: 2017`:

```
    "parserOptions": {
        "ecmaVersion": 2017,
        "sourceType": "module"
    },
```

**Comment by pornel** [comment]
Oh, great! Thank you.

So the real problem here is I guess that the error message gives no hope of this working. It'd be great if it pointed out that it's an ES7 feature used in ES6-only mode.

**Comment by not-an-aardvark** [comment]
To suggest a more descriptive parsing error, I would recommend leaving an issue [here](https://github.com/ternjs/acorn/issues), since that's the parser that ESLint uses by default.
